lgsm/functions/check_steamcmd.sh

```diff
@@ -63,6 +63,43 @@ fn_check_steamcmd(){
 	fi
 }
 
+fn_check_steamcmd_dir(){
+	# Worksround that pre-installs the correct steam directories to ensure all packages use the correct Standard
+	# https://github.com/ValveSoftware/steam-for-linux/issues/6976#issuecomment-610446347
+	if [ ! -d "${XDG_DATA_HOME:="${HOME}/.local/share"}" ]; then
+		mkdir -p "${XDG_DATA_HOME}"
+	fi
+
+	if [ ! -d "${HOME}/.steam" ]; then
+		mkdir -p "${HOME}/.steam"
+	fi
+
+	if [ ! -L "${HOME}/.steam/root" ]; then
+		if [ -d "${HOME}/.steam/root" ]; then
+			rm "${HOME}/.steam/root"
+		fi
+		ln -s "${XDG_DATA_HOME}/Steam" "${HOME}/.steam/root"
+	fi
+
+	if [ ! -L "${HOME}/.steam/steam" ]; then
+		if [ -d "${HOME}/.steam/steam" ]; then
+			rm "${HOME}/.steam/steam"
+		fi
+		ln -s "${XDG_DATA_HOME}/Steam" "${HOME}/.steam/steam"
```

Comment:
@smcv Just wanted to check this is what you mean.

Reply:
When I mentioned `~/.local/share/Steam` on the Steam bug, I was using it as shorthand for what the XDG Base Directory specification says:

* if `XDG_DATA_HOME` is set to a non-empty value, use`${XDG_DATA_HOME}/Steam` as the default for a new Steam installation directory
* if `XDG_DATA_HOME` isn't set or is set to an empty value, use `~/.local/share/Steam` as the default for a new Steam installation directory

The canonical reference for where Steam gets installed is `bin_steam.sh` in a full Steam installation. You can get a relatively up-to-date copy from the Steam launcher/bootstrapper, which is available from https://repo.steampowered.com/steam/archive/precise/ (download and unpack `steam_latest-beta.tar.gz` or `steam_latest.tar.gz` - currently the latest beta is 1.0.0.62 and the latest stable release is 1.0.0.61).

There are two common ways to implement `XDG_DATA_HOME` correctly in shell script:

1. Whenever you want to know the XDG base directory for data, use `${XDG_DATA_HOME:-"${HOME}/.local/share"}`. The `:-` means: if the variable `XDG_DATA_HOME` is set to a non-empty value, "return" that value. Otherwise, "return" the result of expanding the `HOME` variable in `"${HOME}/.local/share"`. This is how `bin_steam.sh` does it. Or:

2. Near the beginning of your script, use `${XDG_DATA_HOME:="${HOME}/.local/share"}` (typically as a parameter to a `:` or `true` so that it doesn't do anything). The `:=` means: if the variable `XDG_DATA_HOME` is set to a non-empty value, "return" that value. Otherwise, *set the variable* to the result of expanding the `HOME` variable in `"${HOME}/.local/share"`, and then "return" the new value. After doing that, you can use `${XDG_DATA_HOME}` unconditionally, because its value has already been set to a default if necessary.

It looks as though in this MR, you're doing a mixture of (1) and (2) by using `${XDG_DATA_HOME:="${HOME}/.local/share"}` everywhere, which is slightly excessive but should also work fine.